code/TagField.php

```diff
@@ -241,108 +231,110 @@ public function saveInto(DataObjectInterface $record) {
 		parent::saveInto($record);
 
 		$name = $this->getName();
-		$relationTitle = $this->getRelationTitle();
+		$titleField = $this->getTitleField();
+
+		/**
+		 * @var DataList $source
+		 */
+		$source = $this->getSource();
 
+		$dataClass = $source->dataClass();
+
+		/**
+		 * @var DataList $values
+		 */
 		$values = $this->Value();
 
-		if(empty($values) || empty($record) || empty($relationTitle)) {
+		if(!$values) {
+			$values = array();
+		}
+
+		if(empty($record) || empty($source) || empty($titleField)) {
 			return;
 		}
 
+		/**
+		 * @var DataObject $record
+		 */
 		if($record->hasMethod($name)) {
```

Comment:
Should this throw an exception if no method can be found? Otherwise the field save won't have any effect, leading to frustrating debugging sessions?